src/lib/parser.c

```diff
@@ -249,7 +257,7 @@ bool handle_attr_event(yaml_event_t *event,
         return push_handler(state, on_seq_scalar_event);
     case YAML_SEQUENCE_END_EVENT:
         /* XXX we know that sequences never come first so the previous state (map) has the key */
-        assert(state->s);
+        assert_analyzer(state->s);
```

Comment:
Where is this one coming from, is --debug performed at configure time? The assert() goes away if NDEBUG is defined, and --debug control's that. Where is this getting generated?

Reply:
you are correct, scan-build runs ok with --enable-debug... hmpf...

Reply:
Iv'e been burned so many times going, WTH... and then going oh yeah its not a debug build so the asserts are gone.